src/fuzzaldrin.cc

```diff
@@ -47,7 +47,23 @@ Napi::Value Fuzzaldrin::SetCandidates(const Napi::CallbackInfo &info) {
     return Napi::Boolean();
 }
 
-void Fuzzaldrin::SetCandidates(const vector<CandidateObject> &candidates) {
+Napi::Value Fuzzaldrin::setTreeFiltererCandidates(const Napi::CallbackInfo &info) {
+    // parse arguments
+    if (info.Length() != 4
+        || !info[0].IsArray()
+        || !info[1].IsString() || !info[2].IsString()) {
+        Napi::TypeError::New(info.Env(), "Invalid arguments").ThrowAsJavaScriptException();
+        return Napi::Boolean();
+    }
+    const auto jsTreeArray = info[0].As<Napi::Array>();
+    const string dataKey = info[1].As<Napi::String>();
+    const string childrenKey = info[2].As<Napi::String>();
+
+    // create Tree and set candidates
+    _tree = Tree(jsTreeArray, dataKey, childrenKey);
```

Comment:
Need an assignment operator constructor

Reply:
Fixed in https://github.com/atom-ide-community/fuzzaldrin-plus-fast/pull/42/commits/a4cc2abedb877a5501df5a7f0633d03f9dc2bc01